TC_S_046 - AssertionError: None != 900

### Information about bug

======================================================================
FAIL: test_sales_order_create_si_via_pe_dn_with_pricing_rule_TC_S_046 (erpnext.selling.doctype.sales_order.test_sales_order.TestSalesOrder)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/frappe-bench/apps/erpnext/erpnext/selling/doctype/sales_order/test_sales_order.py", line 4188, in test_sales_order_create_si_via_pe_dn_with_pricing_rule_TC_S_046
    self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': pe.name, 'account': 'Debtors - _TC'}, 'credit'), 900)
AssertionError: None != 900

----------------------------------------------------------------------
Ran 1 test in 3.532s

FAILED (failures=1)

### Module

other

### Version

Assets: v0.0.1 (pre-prod)

ERPNext: v15.58.2 (pre-prod)

ERPNext CRM: v0.0.1 (pre-prod)

ERPNext India: v0.0.1 (pre-prod)

Frappe Framework: v15.63.1 (pre-prod)

India Compliance: v15.17.1 (pre-prod)

Payments: v0.0.1 (version-15)

Projects: v0.0.1 (pre-prod)

Sales Commission: v0.0.1 (pre-prod)

### Installation method

None

### Relevant log output / Stack trace / Full Error Message.

```shell

```

--- a/erpnext/selling/doctype/sales_order/test_sales_order.py
+++ b/erpnext/selling/doctype/sales_order/test_sales_order.py
@@ -4614,9 +4614,14 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 		self.assertEqual(si.status, "Paid")
 
 	def test_sales_order_create_si_via_pe_dn_with_pricing_rule_TC_S_046(self):
+		self.customer = frappe.get_doc("Customer", self.customer)
+		# Append company if not already present
+		company = "_Test Company"
+		if company and not any(c.company == company for c in self.customer.companies):
+			self.customer.append("companies", {"company": company})
+			self.customer.save()
 		make_item_price()
 		make_pricing_rule()
-
 		so = make_sales_order(qty=10, rate=90)
 		so.save()
 		so.submit()

